src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -50,8 +50,8 @@ public async Task AddAuthCodeAsync()
                 if (csProjFiles.Count() != 1)
                 {
                     var errorMsg = string.Format(Resources.ProjectPathError, _toolOptions.ProjectFilePath);
-                    _consoleLogger.LogFailure(errorMsg, Commands.UPDATE_PROJECT_COMMAND);
-                    return;
+                    _consoleLogger.LogFailure(errorMsg);
+                    Environment.Exit(1);
```

Comment:
more of a question but thinking the reason why we are not adding `Environment.Exit` to `LogFailure` is I imagine we need to log failure without exiting in some instances? If not, we can probably move it in. Haven't used LogFailure as much as you have.

Reply:
I think this is a good point, I'm refactoring LogFailure to LogFailureAndExit and including the Environment.Exit there

Reply:
sounds good.